crates/chain/src/tx_graph.rs

```diff
@@ -343,14 +343,13 @@ impl<A> TxGraph<A> {
         let outputs_sum = tx
             .output
             .iter()
-            .map(|txout| txout.value.to_sat() as i64)
-            .sum::<i64>();
+            .map(|txout| txout.value.to_signed().expect("valid `SignedAmount`"))
+            .sum::<SignedAmount>();
 
         let fee = inputs_sum - outputs_sum;
-        if fee < 0 {
-            Err(CalculateFeeError::NegativeFee(fee))
-        } else {
-            Ok(fee as u64)
+        match fee.to_unsigned() {
+            Ok(fee) => Ok(fee),
+            Err(_) => Err(CalculateFeeError::NegativeFee(fee)),
         }
```

Comment:
This can be done more concisely like this: 
```suggestion
        fee.to_unsigned()
            .map_err(|_| CalculateFeeError::NegativeFee(fee))
```

Reply:
Thanks, that's better, updated